### PR TITLE
Devise related branch

### PR DESCRIPTION
**Pull request by Terunaga**
# WHAT

Set up some initial actions in order to implement the sign-up features.
Specifically, ran the following commands:
- rails g devise:install
- rails g devise user
  *rails g devise:views

Also, added the following columns to the users table:
- username
- member
- profile
- works
# WHY

These setups are crucial in implementing the sign-up features.

### CODE REVIEW

File: app/views/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %> 
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (sinagaki58) [inline app/views/devise/confirmations/new.html.erb]: erbは全てhamlにしましょう
erb2hamlというgemがあります

Comment (sinagaki58) [inline app/views/devise/confirmations/new.html.erb]: 空行も

File: app/views/devise/unlocks/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend unlock instructions" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (sinagaki58) [inline app/views/devise/unlocks/new.html.erb]: deviseのview全て空行いれましょう

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (sinagaki58) [inline test/models/user_test.rb]: test消しましょう

**Comment by sinagaki58** [comment]
作業中のブランチだと思うので、その場合は
`[WIP]Devise related branch`
みたいな感じで[WIP]とつけましょう

**Comment by Terunaga** [comment]
# WHAT
- Converted all the view files to html.haml extensions with rake haml:erb2haml command.
- Added a blank line to each view file created.

# WHY
- The view files will be created by html.haml extension.
- It gives a trouble without a blank line at the end of a file.

**Comment by sinagaki58** [comment]
lgtm

**Comment by Terunaga** [comment]
# WHAT

##【WIP】Devise related branch
- added gem "carrierwave".
- implemented user sign-up/sign-in/sign-out functions.
- added contents to update action of user controller.
- modified the view files for "my page", "edit', "sign in", "sign up" pages.

# WHY
- carrierwave gem is necessary for adding an avatar.
- a user can experience different features depending on whether a user has logged in or not.
- a user can edit his/her profile.
- some links were not added/haml wasn't functioning properly, etc.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -2,4 +2,15 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.for(:sign_up).push(:username, :avatar, :member, :profile, :works)
+  end
```
Comment (sinagaki58) [inline app/controllers/application_controller.rb:10]: account_updateも必要ですね

File: app/controllers/user_controller.rb [hunk 5]
```diff
@@ -10,12 +10,20 @@ def edit
   end
 
   def show
+    @user = User.find(params[:id])
   end
 
   def update
```
Comment (sinagaki58) [inline app/controllers/user_controller.rb:10]: editアクションも必要かと

File: app/views/layouts/_header.html.haml [hunk 6]
```diff
@@ -4,25 +4,26 @@
...
+              %li
+                = link_to 'Logout', destroy_user_session_path, method: :delete
+          %li
+            .btn.btn-primary.navbar-btn
+              = link_to 'New Proto', new_prototype_path
+      - else
+        %ul.nav.navbar-nav.navbar-right
+          %li
+            .btn.btn-primary.navbar-btn
+              = link_to 'Get Started', new_user_session_path
```
Comment (sinagaki58) [inline app/views/layouts/_header.html.haml]: 空行

File: app/views/user/show.html.haml [hunk 7]
```diff
@@ -2,8 +2,8 @@
   %header.user-nav
     .media
       .media-left
-        %a{href: "#"}
-          %img.media-object{alt: "64x64", data: {holder: {rendered: true}}, data: {src: "holder.js/64x64"}, src: "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9InllcyI/PjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgcHJlc2VydmVBc3BlY3RSYXRpbz0ibm9uZSI+PGRlZnMvPjxyZWN0IHdpZHRoPSI2NCIgaGVpZ2h0PSI2NCIgZmlsbD0iI0VFRUVFRSIvPjxnPjx0ZXh0IHg9IjEzLjQ2MDkzNzUiIHk9IjMyIiBzdHlsZT0iZmlsbDojQUFBQUFBO2ZvbnQtd2VpZ2h0OmJvbGQ7Zm9udC1mYW1pbHk6QXJpYWwsIEhlbHZldGljYSwgT3BlbiBTYW5zLCBzYW5zLXNlcmlmLCBtb25vc3BhY2U7Zm9udC1zaXplOjEwcHQ7ZG9taW5hbnQtYmFzZWxpbmU6Y2VudHJhbCI+NjR4NjQ8L3RleHQ+PC9nPjwvc3ZnPg==", style: "width: 64px; height: 64px;"}/
+        %a{href: user_path(current_user)}
```
Comment (sinagaki58) [inline app/views/user/show.html.haml]: aタグはlink_toにしましょう

**Comment by sinagaki58** [comment]
画像はpushせずにgitignoreに入れましょう

**Comment by Terunaga** [comment]
# WHAT
- modified <a> tag to link_to method.
- deleted unnecessary actions in user controller.
- added a blank line.
- put public/uploads folder in .gitignore.

# WHY
- link_to method is better than <a> tag.
- some actions are not needed.
- it gives a trouble without a blank line.
- to prevent pushing avatars.

**Comment by sinagaki58** [comment]
pushした画像消したらlgtmです

**Comment by Terunaga** [comment]
@sinagaki58 
Git revertで画像を消せたみたいです。
確認お願いします！

**Pull request merged**